File: backend/src/main/java/com/silyosbekov/chessmate/dto/MoveDto.java
package com.silyosbekov.chessmate.dto;

import com.silyosbekov.chessmate.model.PlayerColor;

import java.util.UUID;

public record MoveDto(
        UUID gameId,
        UUID whitePlayerId,
        UUID blackPlayerId,
        PlayerColor color,
        String from,
        String to,
        boolean isCheckmate,
        boolean isStalemate)
{
}

File: frontend/projects/chessmate-app/src/app/core/models/game/moveDto.ts
import {PlayerColor} from './playerColor';

export interface MoveDto {
  gameId: string;
  whitePlayerId: string;
  blackPlayerId: string;
  color: PlayerColor;
  from: string;
  to: string;
  isCheckmate: boolean;
  isStalemate: boolean
  san: string;
  pgn: string;
}
